sql/core/src/main/scala/org/apache/spark/sql/catalog/Catalog.scala

```diff
@@ -342,6 +342,44 @@ abstract class Catalog {
     createTable(tableName, source, schema, options)
   }
 
+  /**
+   * Creates a table based on the dataset in a data source and a set of options.
+   * Then, returns the corresponding DataFrame.
+   *
+   * @param tableName is either a qualified or unqualified name that designates a table.
+   *                  If no database identifier is provided, it refers to a table in
+   *                  the current database.
+   * @since 3.1.0
+   */
+  def createTable(
+      tableName: String,
+      source: String,
+      options: java.util.Map[String, String],
+      description: String): DataFrame = {
+    createTable(
+      tableName,
+      source = source,
+      options = options.asScala.toMap,
+      description = { if (description.isEmpty) None else Some(description) }
+    )
+  }
+
+  /**
+   * (Scala-specific)
+   * Creates a table based on the dataset in a data source and a set of options.
+   * Then, returns the corresponding DataFrame.
+   *
+   * @param tableName is either a qualified or unqualified name that designates a table.
+   *                  If no database identifier is provided, it refers to a table in
+   *                  the current database.
+   * @since 3.1.0
+   */
+  def createTable(
+      tableName: String,
+      source: String,
+      options: Map[String, String],
+      description: Option[String]): DataFrame
```

Comment:
I think we don't need to make the `description` `Option`al. We can just call:

```scala
  override def createTable(
      tableName: String,
      source: String,
      schema: StructType,
      options: Map[String, String]): DataFrame = {
    createTable(tableName, source, schema, options, None)
  }
```

if `description` is not specified in Python side.

Reply:
Python won't call this particular signature.

I assume you meant to make this comment on the signature that includes `java.util.Map[String, String]`, right? `description` is already a plain `String` there. The `Option[String]` is just for the Scala API.

Does that address your concern, or am I misunderstanding something?

Reply:
Just to be clear about what I think I'm doing here (because I'm unsure of myself and could use your confirmation):

At a high level, we have two versions of the `createTable()` APIs: one for Scala, and one for Java and Python. Python calls the Java versions of the API via Py4J.

On the Scala side, it's OK to use `Option[String]`. On the Java/Python side, we need to use a plain `String` and then convert that to an `Option[String]` when calling from the Java side into the Scala. That's what I'm currently doing.

Am I understanding correctly?

Reply:
Hmmmm .. why should this be `Option` alone in Scala? I thought it's better to keep the signature close to each other.